Block/Js.php

```diff
@@ -354,13 +354,19 @@ private function isPageRestricted()
             return true;
         }
 
-        // If minicart is supported (allowing Bolt on every page)
+        // If minicart/product page checkout is supported (allowing Bolt on every page)
         // and no IP whitelist is defined there are no additional restrictions.
-        if ($this->configHelper->getMinicartSupport() && !$this->configHelper->getIPWhitelistArray()) {
-            return false;
+        if (!$this->configHelper->getIPWhitelistArray()) {
+            if ($this->configHelper->getMinicartSupport()) {
+                return false;
+            }
+            if ($type=="withproductpage" && ($currentPage=="catalog_product_view") && $this->configHelper->getProductPageCheckoutFlag()) {
```

Comment:
I don't think $type is necessary. "catalog_product_view" should be enabled by default and added to the default whitelist. Should the JS load can be determined at the begining of the phtml file (getProductPageCheckoutFlag), as done with other files.